processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/EndRangeValidatorGenerator.java

```diff
@@ -147,8 +147,8 @@ private static CodeBlock generateNoticeContext(
     TypeName tableLoaderTypeName = new GtfsEntityClasses(fileDescriptor).tableLoaderTypeName();
     CodeBlock.Builder block =
         CodeBlock.builder().add("$T.FILENAME, entity.csvRowNumber(), ", tableLoaderTypeName);
-    if (fileDescriptor.primaryKey().isPresent()) {
-      block.add("entity.$L(), ", fileDescriptor.primaryKey().get().name());
+    if (fileDescriptor.hasSingleColumnPrimaryKey()) {
+      block.add("entity.$L(), ", fileDescriptor.getSingleColumnPrimaryKey().name());
```

Comment:
I think there may be some other notices that assume a single-column primary key.  Fixing something like this would likely require a refactor of the individual notices, but I suppose that would be a breaking change for users who assume the notice schema is fixed?  @maximearmstrong do you have any thoughts on refactoring notice schemas?

Reply:
Refactoring the notice schemas would be a breaking change indeed. Let's avoid it if we can.

Though, I'm not sure to understand why single column primary key entities could be a problem. Could you elaborate on this?

Reply:
It's only a problem for multi-column primary keys that must be discussed in notices that only support single columns.

Reply:
I understand. If you think this refactor would valuable in the long run, then we should plan it. I think it would be part v4. Though, I think your changes make sense for now. cc @isabelle-dr